File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/InventoryItem.swift
import Foundation

// MARK: - InventoryType

public enum InventoryType: String, Codable, Sendable, CaseIterable, Hashable {
    case ingredient
    case equipment
    case supply
}

// MARK: - InventoryItem

public struct InventoryItem: Codable, Identifiable, Sendable, Hashable {
    public let id: String
    public let userId: String
    public let ingredientName: String
    public let currentStock: Double
    public let minimumStock: Double
    public let unit: String
    public var unitCost: Double?
    public let lastUpdated: String
    public let type: InventoryType

    public init(
        id: String,
        userId: String,
        ingredientName: String,
        currentStock: Double,
        minimumStock: Double,
        unit: String,
        unitCost: Double? = nil,
        lastUpdated: String,
        type: InventoryType
    ) {
        self.id = id
        self.userId = userId
        self.ingredientName = ingredientName
        self.currentStock = currentStock
        self.minimumStock = minimumStock
        self.unit = unit
        self.unitCost = unitCost
        self.lastUpdated = lastUpdated
        self.type = type
    }
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/InventoryItem.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class InventoryItem(
    val id: String = "",
    @SerialName("user_id") val userId: String = "",
    @SerialName("ingredient_name") val ingredientName: String,
    @SerialName("current_stock") val currentStock: Double,
    @SerialName("minimum_stock") val minimumStock: Double,
    val unit: String,
    @SerialName("unit_cost") val unitCost: Double? = null,
    @SerialName("last_updated") val lastUpdated: String = "",
    val type: InventoryType
)

@Serializable
enum class InventoryType {
    @SerialName("ingredient") INGREDIENT,
    @SerialName("equipment") EQUIPMENT,
    @SerialName("supply") SUPPLY
}
